src/sql/platform/connection/common/connectionConfig.ts

```diff
@@ -104,36 +104,34 @@ export class ConnectionConfig {
 	/**
 	 *Returns group id
 	 */
-	public addGroupFromProfile(profile: IConnectionProfile): Promise<string> {
+	public addGroupFromProfile(profile: IConnectionProfile): Promise<IConnectionProfileGroup> {
 		if (profile.groupId && profile.groupId !== Utils.defaultGroupId) {
-			return Promise.resolve(profile.groupId);
-		} else {
+			return Promise.resolve(this.getAllGroups().find(g => g.id === profile.groupId)!);
+		} else if (profile.groupFullName) {
```

Comment:
Why are you changing this behavior? Root doesn't have a groupFullName so it's going to throw an error for that now

Reply:
name is a required field for the connectiongroupshape, so if we don't have a profile groupname we can't do anything here.

Reply:
Unless the assumption is if there is no group name, its the default group? If so I can fix this and add a comment regarding that. (This is not very well defined).

Reply:
Utils.defaultGroupId is what is used to determine if it's actually in the root group - although I'm pretty sure we could also take a blank name as being the default group. The logic currently works that way but I'm not sure it's actually on purpose. 

Looking at this more this whole area seems pretty sketchy - for example why do we have two separate methods addGroup and addGroupFromProfile that handle adding a group differently?

At this point I'd want to do more investigation so for purposes of this change I'd prefer we try to keep the logic intact such that it does the same thing it did previously with undefined/empty names.

Reply:
https://github.com/microsoft/azuredatastudio/blob/master/src/sql/platform/connection/common/connectionProfileGroup.ts#L188

So yeah this check here says it's the root if the name is blank or is ROOT (which is a bug actually - you can't have a group named root because it thinks it already exists. Hooray.)

Reply:
TBH I just made these changes because there was some strict-null-check stuff here, but I can just revert all those changes and just update the error counts. Whoever tackles this area for strict-null-checks can fix this issues throughout during that PR.

Reply:
Yeah let's just do that. The chance of breaking something here is too high (especially in an already decently large PR)